packages/xrpl/src/sugar/submit.ts

```diff
@@ -70,18 +75,25 @@ async function submitAndWait(
     failHard?: boolean
     // A wallet to sign a transaction. It must be provided when submitting an unsigned transaction.
     wallet?: Wallet
+    // Custom rippled types to use instead of the default. Used for sidechains and amendments.
+    definitions?: XrplDefinitions
   },
 ): Promise<TxResponse> {
   const signedTx = await getSignedTx(this, transaction, opts)
 
-  const lastLedger = getLastLedgerSequence(signedTx)
+  const lastLedger = getLastLedgerSequence(signedTx, opts?.definitions)
```

Comment:
Why does this method need `definitions`?

Reply:
In order to decode the transaction - if it happens to be a new type, then `decode` would fail otherwise.